### PR TITLE
Feature/enhance powerup entities

### PR DESCRIPTION
**Pull request by sarahsturgeon**
~~Opened on behalf of~~ Stolen from Peri

### CODE REVIEW

File: moon/entities/base_cfc_powerup.moon [hunk 1]
```diff
@@ -62,5 +74,10 @@ ENT.GivePowerup = (ply) =>
...
 
-ENT.Use = (activator) =>
-    @GivePowerup activator
+ENT.PhysicsCollide = (colData, collider) =>
+    collideEnt = colData.HitEntity
+
+    isValidPlayer = IsValid(collideEnt) and collideEnt\IsPlayer!
+
+    if isValidPlayer
+        @GivePowerup collideEnt
```
Comment (brandonsturgeon) [inline moon/entities/base_cfc_powerup.moon:83]: I'd suggest reversing the case there, but it's up to you!

```suggestion
    if not isValidPlayer then return end
    @GivePowerup collideEnt
```

File: moon/utils/server/sv_powerups_spawner.moon [hunk 2]
```diff
@@ -43,6 +43,14 @@ PowerupSpawner =
 
         spawnLocations
 
+    spawnPowerup: (powerupClass, position) ->
+        powerup = ents.Create powerupClass
+        powerup.originalPos = position
+        powerup\SetPos position
+        powerup\Spawn!
+
+        powerup\EmitSound "ambient/machines/teleport4.wav", 90
```
Comment (sarahsturgeon) [inline moon/utils/server/sv_powerups_spawner.moon]: This should be made configurable, maybe by convar?

File: moon/powerups/server/viper.moon [hunk 3]
```diff
@@ -44,20 +42,22 @@ class ViperPowerup extends BasePowerup
...
-            dmg\ScaleDamage MELEE_DAMAGE_MULT
+            multiplier = getConf "viper_multiplier"
+            dmg\ScaleDamage multiplier
 
     ApplyEffect: =>
         @owner\SetColor PLAYER_COLOR
 
-        timer.Create @timerName, POWERUP_DURATION, 1, -> @Remove!
+        duration = getConf "viper_duration"
+        timer.Create @timerName, duration, 1, @Remove
```
Comment (samuelWilliams99) [inline moon/powerups/server/viper.moon]: This compiles to `timer.Create(self.timerName, duration, 1, self.Remove)`, not passing `self` into `Remove`
```suggestion
        timer.Create @timerName, duration, 1, -> @Remove!
```

Comment (sarahsturgeon) [reply moon/powerups/server/viper.moon]: Ah, knew there was a reason I did it that way

File: moon/powerups/server/regen.moon [hunk 4]
```diff
@@ -1,49 +1,52 @@
...
         @timerName = "CFC_Powerups-Regen-#{ply\SteamID64!}"
 
-        timerDuration = POWERUP_DURATION / REGEN_INTERVAL
-        timer.Create @timerName, REGEN_INTERVAL, timerDuration, @PowerupTick!
+        duration = getConf "regen_duration"
+        interval = getConf "regen_interval"
 
-        @RegenSound = CreateSound(@owner, REGEN_SOUND)
+        timerDuration = duration / interval
+        timer.Create @timerName, interval, timerDuration, @PowerupTick!
```
Comment (ghost) [inline moon/powerups/server/regen.moon]: Pretty sure you don't call the function when creating a timer, also can't use the @ reference in a timer I think
```suggestion
        timer.Create @timerName, interval, timerDuration, -> @PowerupTick!
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — COMMENTED** [review]
Looks good to me 👍

**Pull request merged**